elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/GraphQLDSL.java

```diff
@@ -0,0 +1,585 @@
+        return new VariableValue(name);
+    }
+
+    /**
+     * Creates a query spec that represents a object value.
+     * <p>
+     * For example, "id:\"1\", title:\"update title\"" as in "{ id:\"1\", title:\"update title\" }".
+     *
+     * @param object  A string representation of the object wrapped
+     *
+     * @return a GraphQL spec
+     */
+    public static ValueWithVariable objectValueWithVariable(Object object) {
+        return new ObjectValueWithVariable(object);
+    }
+
+    /**
+     * Creates a string value that is NOT quoted in its serialized form
+     *
+     * @param enumValue  The same string in the serialized form of the created object.
+     *
+     * @return a GraphQL spec
+     */
+    public static ValueWithVariable enumValue(String enumValue) {
+        return new EnumValue(enumValue);
+    }
+
+    /**
+     * Creates an response attribute(scalar field).
+     *
+     * @param name  The name of the selected attribute that would appear in a GraphQL query.
+     * @param value  The attribute value
+     *
+     * @return a field that represents an non-reltionship entity response attribute
+     */
+    public static Selection responseField(String name, String value) {
+        return com.yahoo.elide.contrib.testhelpers.graphql.elements.response.ScalarField.withQuotedValue(name, value);
+    }
+
+    /**
+     * Creates an response attribute(scalar field) whose value is {@code null}(unquoted).
+     *
+     * @param name  The name of the selected attribute that would appear in a GraphQL query.
+     *
+     * @return a field that represents an non-reltionship entity response attribute whose value is
+     * {@code null}(unquoted)
+     */
+    public static Selection nullResponseField(String name) {
+        return com.yahoo.elide.contrib.testhelpers.graphql.elements.response.ScalarField.withUnquotedValue(name, null);
+    }
+
+    /**
+     * Creates an response attribute(scalar field) whose value is a GSON-serializable object, such as map.
+     *
+     * @param name  The name of the selected attribute that would appear in a GraphQL query.
+     *
+     * @return a field that represents an non-reltionship entity response attribute whose value a GSON-serializable
+     * object, such as map.
+     */
+    public static Selection jsonResponseField(String name, Object jsonOjbect) {
+        return com.yahoo.elide.contrib.testhelpers.graphql.elements.response.ScalarField.withUnquotedValue(
+                name,
+                GSON_INSTANCE.toJson(jsonOjbect)
+        );
+    }
+
+    /**
+     * Creates a entity(object field) response attribute.
+     *
+     * @param name  The name of the selected entity/field that would appear in a GraphQL query
+     * @param nodes  All instances of the entity in the response data
+     *
+     * @return an entity selection instantiation
+     */
+    public static Selection responseField(String name, SelectionSet... nodes) {
```

Comment:
Is it possible to collapse responseField into field as well?  Again - fewer concepts.  If you are concerned about serialization with and without quotes, maybe that can just be an option to the DSL itself.

Reply:
Done